aws/http_client.go

```diff
@@ -110,51 +91,49 @@ func (b BuildableHTTPClient) initReset() BuildableHTTPClient {
 // If a non (*http.Transport) was set as the round tripper, the round tripper
 // will be replaced with a default Transport value before invoking the option
 // functions.
-func (b BuildableHTTPClient) WithTransportOptions(opts ...func(*http.Transport)) HTTPClient {
-	b = b.initReset()
+func (b *BuildableHTTPClient) WithTransportOptions(opts ...func(*http.Transport)) HTTPClient {
+	cpy := b.clone()
 
-	tr := b.GetTransport()
+	tr := cpy.GetTransport()
```

Comment:
Should this still b `b`? Wouldn't we want to copy the existing transport that might have been set?